api/lcp/controller.py

```diff
@@ -54,21 +54,21 @@ def _get_lcp_passphrase(self, patron):
 
         return lcp_passphrase
 
-    def _get_lcp_collection(self, library):
+    def _get_lcp_collection(self, patron, collection_name):
         """Returns an LCP collection for a specified library
         NOTE: We assume that there is only ONE LCP collection per library
 
-        :param library: Library
-        :type library: core.model.library.Library
+        :param patron: Patron object
+        :type patron: core.model.patron.Patron
+
+        :param collection_name: Name of the collection
+        :type collection_name: string
 
         :return: LCP collection
         :rtype: core.model.collection.Collection
         """
-        lcp_collection = next(iter([
-            collection
-            for collection in library.collections
-            if collection.protocol == ExternalIntegration.LCP
-        ]))
+        db = Session.object_session(patron)
+        lcp_collection, _ = Collection.by_name_and_protocol(db, collection_name, ExternalIntegration.LCP)
```

Comment:
This allows the possibility that you can authenticate an HTTP request against a collection that's in someone else's library, if you know the name of a collection. If `lcp_collection` is not in `patron.library.collections` this method should return a problem detail document.

Reply:
Sorry, missed that. I added a check below as requested